### PR TITLE
Conditionals and animation

### PR DESCRIPTION
**Pull request by zsmoore**
#190 
#189 

# Description

This PR finishes up visualization for conditionals and hooks it up with the interpreter rep.

# Change Log
Describe all changes made 

All bad interpreter code is marked with hack or kept in an extra file.
Visualizer code will be reusable in the future and sits in the components

# Manual Test Steps
Test conditionals as well as the visual output

### CODE REVIEW

**Review by zsmoore — COMMENTED** [review]
Going to give a +1 to Noah's portion of the code

**Review by ghost — APPROVED** [review]

File: src/view/visualizer/sketch/components/conditionalView.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    this.possibilities = conditional.possibilities;
+    this.chosen = conditional.chosen; // index of the possibility that gets executed
+    this.isAnimating = false;  // currently executing the scope swiping animation
+    this.curFrame = 0;
+
+    // create animation keyframes
+    this.numFrames = 100;
+    const curve = new Bezier(3, 0, 0, 0, 3, 1, 0, 1);
+
+    console.log('Computing animation path...');
```
Comment (angushtlam) [inline src/view/visualizer/sketch/components/conditionalView.js]: remove pls

File: src/view/visualizer/sketch/components/conditionalView.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      if (Math.abs(delta) < minDistance) {
+        minDistance = delta;  // still approaching
+      } else { // once distance starts growing again, we've overshot the point. this is it.
+        oldTime = t;
+        minDistance = curveLength;
+        steps.push(t - tstep); // found our point, on to the next one
+      }
+    }
+
+    console.log('Complete!');
```
Comment (angushtlam) [inline src/view/visualizer/sketch/components/conditionalView.js]: remove pls

**Review by angushtlam — CHANGES_REQUESTED** [review]
jk

**Review by angushtlam — APPROVED** [review]

**Comment by zsmoore** [comment]
@angushtlam 
Will fix during re-work

**Pull request merged**